### ISSUE TITLE
Use a numerical quadrature (simpson should do) to approximate the length of the hearth at a specified time

### ISSUE THREAD

**Issue by eudaldrg** [gloriamacia/Cardiac-Matlab-Study#3]

**Comment by gloriamacia** [comment]
Oh, interesting. If I understood it correctly right now I was just interpolating the lenght by assuming the distance between two points an straight line. This new methodology would be more appropiate for the ventricle rounded shape, right? Thanks in advance!

**Comment by eudaldrg** [comment]
Yes! It's similar to the point interpolation techniques. When you rise the degree of the interpolation polinomial you reduce the error. This is the base idea but you cannot keep increasing the degree indefinetely because other problems arise (google Runge Phenomenon). When you use polinomials to compute integrals (like when you compute the ventricle length) the idea is pretty similar.
If, at some point, the shape is not smooth enough, the error at these points can increase a lot (if you want further information, tell me). Furthermore, when increasing the order of the polinomial, new interpolation formulas have to be deduced. Because of that, the usual methods use first or second degree polinomials and divide the points in groups of 2 or 3 and apply the interpolation there. You did it for the parallelogram method, which uses a first degree polinomial. Simpsons uses a 2nd degree one

**Comment by gloriamacia** [comment]
Thank you so much! I really liked your contribution.

**Comment by eudaldrg** [comment]
I found a Matlab function that automatically chooses a good quadrature, my local version uses it now. Even if the program uses this function, I'll also push a handmade version to let the user choose between different quadrature approaches to compare them and see how they work

**Comment by gloriamacia** [comment]
This is a really good idea. Thanks for your contributions!
